### PR TITLE
chore(new_chart): Add Chaos Charts for pod autoscaler experiment

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: Udit Gaurav <udit.gaurav@mayadata.io>

**This PR consist:**

- New Chart for pod autoscaler experiment

### CODE REVIEW

File: charts/generic/pod-autoscaler/pod-autoscaler.chartserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  name: pod-autoscaler
+  version: 0.1.0
+  annotations:
+    categories: Kubernetes
+    vendor: CNCF
+    support: https://slack.kubernetes.io/
+spec:
+  displayName: pod-autoscaler
+  categoryDescription: |
+    The experiment aims to check the ability of nodes to accommodate the number of replicas of the same application pod.
```
Comment (ksatchit) [inline charts/generic/pod-autoscaler/pod-autoscaler.chartserviceversion.yaml]:
```suggestion
    The experiment aims to check the ability of nodes to accommodate the number of replicas a given application pod.
```

File: charts/generic/pod-autoscaler/pod-autoscaler.chartserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    Scale- this experiment can be used for different scenarios as well, like for checking the Node auto-scaling feature. If the pods are unable to schedule then the nodes are getting auto-scaled or not.
```
Comment (ksatchit) [inline charts/generic/pod-autoscaler/pod-autoscaler.chartserviceversion.yaml]:
```suggestion
    This experiment can be used for other scenarios as well, such as for checking the Node auto-scaling feature. For example, check if the pods are successfully rescheduled within a specified period in cases where the existing nodes are already running at the specified limits. 
```

File: charts/generic/pod-autoscaler/pod-autoscaler.chartserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  keywords:
+    - Kubernetes
+    - Scale
+    - State
+    - Node
+    - Replicas
+  platforms:
+    - GKE
+    - EKS
+    - AKS
```
Comment (ksatchit) [inline charts/generic/pod-autoscaler/pod-autoscaler.chartserviceversion.yaml]: Let's confirm before we add AKS (should work, but would help understand impact/other implications specific to the platform). We can go w/ the other tested platforms

File: charts/generic/pod-autoscaler/rbac.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: pod-autoscaler-sa
+  namespace: default
+  labels:
+    name: pod-autoscaler-sa
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
```
Comment (ksatchit) [inline charts/generic/pod-autoscaler/rbac.yaml:11]: Can be done at namespaced level. Doesn't really need a cluster role ?

**Review by ksatchit — DISMISSED** [review]

**Review by ksatchit — APPROVED** [review]

**Review by ispeakc0de — APPROVED** [review]

**Pull request merged**